app/client-app/src/app/pages/Editor/components/Webhook/WebhookForm.jsx

```diff
@@ -33,9 +33,122 @@ import LightHeading from "components/core/LightHeading";
 import { generateWebhookSecret } from "api/webhooks";
 import ConfirmationModal from "./ConfirmationModal";
 
-const WebhookForm = ({ isOpen, onClose, onSubmit }) => {
-  const finalRef = useRef(null);
+const CreateModeSecretField = ({
+  values,
+  handleGenerateSecret,
+  setFieldValue,
+}) => {
+  return (
+    <HStack w="100%">
+      <TextField name="secret" placeholder="Secret" header="Header" size="sm" />
+      <Button
+        size="sm"
+        colorScheme="teal"
+        w="40%"
+        onClick={() => handleGenerateSecret(values, setFieldValue)}
+      >
+        Generate Secret
+      </Button>
+    </HStack>
+  );
+};
+
+const SecretField = ({
+  isEditMode,
+  editSecret,
+  handleGenerateSecret,
+  values,
+  setFieldValue,
+  setEditSecret,
+}) => {
+  return (
+    <>
+      {isEditMode ? (
+        <>
+          {isEditMode && (
+            <>
+              {editSecret ? (
+                <>
+                  <Alert status="info" mt={4}>
+                    <VStack pl="2">
+                      <HStack>
+                        <AlertIcon />
+                        <AlertTitle>
+                          Info: Changing Secret May Affect Integrations
+                        </AlertTitle>
+                      </HStack>
+                      <AlertDescription>
+                        To retain a secret, please input a new value. An empty
```

Comment:
⛏️ The "To retain a secret" bit is a little confusing to me. Maybe it's just me, but it implies the old secret may be kept. But the purpose of this component is to change it. I think you can remove the "To retain a secret" bit and start with "Please input a value for the secret." followed by the next sentence